### ISSUE TITLE
Bug: Running fixer only fixes the first error when all errors are reported on the same line/column

### ISSUE THREAD

**Issue by meeroslav** [eslint/eslint#17334]
### Environment

Node version: 20.0.0
npm version: 9.7.1
Local ESLint version: 8.15.0
Global ESLint version: -
Operating System: MacOS

### What parser are you using?

Other

### What did you do?

<details>
<summary>Configuration</summary>

```jsonc
{
  // ...
  "overrides": [
    // ...
    {
      "files": ["*.json"],
      "parser": "jsonc-eslint-parser",
      "rules": {
        "@nx/dependency-checks": "error"
      }
    }
  ]
}
```
</details>

The following code is a snippet from https://github.com/nrwl/nx/blob/master/packages/eslint-plugin/src/rules/dependency-checks.ts. This rule collects the information about all the dependencies of a given package in an Nx monorepo and reports error if some of them are missing in `package.json`. This report is attached to e.g. `dependencies` property of package.json.

Since there might be multiple packages missing, there might be several errors reported on the same node.

```js
context.report({
  node: node,
  messageId: 'missingDependency',
  data: { packageName: d, projectName: sourceProject.name },
  fix(fixer) {
    projPackageJsonDeps[d] = rootPackageJsonDeps[d] || projDependencies[d];
    const deps = node.value.properties;
    if (deps.length) {
      return fixer.insertTextAfter(deps[deps.length - 1], `,\n    "${d}": "${projPackageJsonDeps[d]}"`);
    } else {
      return fixer.insertTextAfterRange(
        [node.value.range[0] + 1, node.value.range[1] - 1], `\n    "${d}": "${projPackageJsonDeps[d]}"\n  `);
    }
  },
});
```

### What did you expect to happen?

When running `eslint --fix` I would expect all of the fixes to be applied. E.g. for several missing dependencies like:
![Screenshot 2023-07-03 at 14 07 09](https://github.com/eslint/eslint/assets/881612/847c60f4-05d2-4d80-8bd7-b96298255718)

I would have expected all dependencies to be added to the `dependencies` section.

### What actually happened?

Applying the `--fix` adds only the first element.

### Link to Minimal Reproducible Example

https://github.com/LayZeeDK/nx-dependency-checks-executor-dependency-issue

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

The workaround is to add all of the missing dependencies in a single fixer or an array of fixers, but I was expecting running and reporting them one by one would work

**Comment by fasttime** [comment]
Hi @meeroslav, I was able to reproduce the problem after removing `"ignorePatterns"` from the [config](https://github.com/LayZeeDK/nx-dependency-checks-executor-dependency-issue/blob/816f76d625f811d64d19e9ab2432db4cf6a83206/.eslintrc.json#L3), because that option was preventing all files from being linted. After this change, I could see that only the first problem reported by the first run of the rule was being fixed, and the next run would not report any problems.

For clarification, ESLint cannot apply multiple *conflicting fixes*: this is stated in the [custom rules documentation](https://eslint.org/docs/latest/extend/custom-rules#conflicting-fixes), but it is not obvious that the same situation occurs when multiple fixes try to insert new text at the same position. In this case, ESLint will apply only one of the fixes, then re-run the rule on the fixed code to see if there are still any fixable problems, and try to fix these ones as well. This process goes on up to 10 times or until the rule stops reporting problems.

So the expectation would be that if a rule finds, say, 12 problems fixable by inserting text at the same position, the fixed file would have 10 problems fixed and 2 unfixed. The issue here is that when the `@nx/dependency-checks` rule runs the second time on a file, it reports no problems at all, even if not all fixes were applied. To me, this points to an issue with the rule itself or within the configuration. My recommendation would be to fix that issue before we start thinking about any further steps in ESLint.
